### PR TITLE
Feat/trajectory tracking

### PR DESCRIPTION
**Pull request by RocketFan**
### Changes:
- fix environment positions and generated trajectory coordinates
- trajectory tracking node with implemented pure pursuit
- trajectory tracking can be parametrized with velocity and radius (look ahead distance)
- trajectory tracking starts when path is sent and send empty message on finish
- add example trajectory tracker roslaunch
- change precision landing end service to topic (same as in trajectory tracking)

### CODE REVIEW

File: src/drone/scan_trajectory.py [hunk 1]
```diff
@@ -11,6 +12,7 @@ def __init__(self, polygon_coords: list, fov_x: float, fov_y: float):
         self.start_point = (0, 0)
         self.fov_x = fov_x
         self.fov_y = fov_y
+        polygon_coords = [(-y, x) for x, y in polygon_coords]
```
Comment (Kaktusowy500) [inline src/drone/scan_trajectory.py:15]: Mixing coordinates order/values here and in environment is a bit confusing, consider refinement

File: src/trajectory/trajectory.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+        self._points = points
+        self._segments = [Segment(start, end) for start, end in zip(points[:-1], points[1:])]
+        self._waypoints = [Waypoint(point, segment) for point, segment in zip(points, self._segments)]
+
+        if len(self.points) > 0:
+            self._waypoints.append(Waypoint(points[-1], self._segments[-1]))
+
+        self._index = 0
+        self._segment = self._segments[0]
```
Comment (Kaktusowy500) [inline src/trajectory/trajectory.py:85]: Yout can probably use `@classmethod` and return class instance instantly

File: src/trajectory_tracker.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    def __init__(self):
+        self.radius = float(rospy.get_param('~radius', 1))
+        self.velocity = float(rospy.get_param('~velocity', 1))
+
+        self.offboard = Offboard()
+        self.trajectory = Trajectory()
+        self.pure_pursiut = PurePursuit(lookahead_distance=self.radius)
+
+        self.sub_trajectory = rospy.Subscriber('trajectory_tracker/path', Path, self.callback_trajectory)
+        self.pub_finished = rospy.Publisher('trajectory_tracker/finished', Empty, queue_size=1)
```
Comment (Kaktusowy500) [inline src/trajectory_tracker.py:35]: Init publisher before subscriber

**Review by Kaktusowy500 — CHANGES_REQUESTED** [review]

File: src/trajectory_tracker.py (hunk 3 above)
Comment (RocketFan) [reply src/trajectory_tracker.py:32]: I changed the order

**Review by Kaktusowy500 — APPROVED** [review]

**Pull request merged**